servers/plugins/org.fusesource.ide.server.fuse.core/src/org/fusesource/ide/server/fuse/core/server/subsystems/Fuse6xStartupLaunchConfigurator.java

```diff
@@ -14,49 +14,31 @@
 import org.fusesource.ide.foundation.core.util.Strings;
 import org.fusesource.ide.server.fuse.core.util.FuseToolingConstants;
-import org.fusesource.ide.server.karaf.core.server.subsystems.BaseKarafStartupLaunchConfigurator;
+import org.fusesource.ide.server.karaf.core.server.subsystems.Karaf2xStartupLaunchConfigurator;
 
 /**
  * @author lhein
  */
-public class Fuse6xStartupLaunchConfigurator extends BaseKarafStartupLaunchConfigurator {
+public class Fuse6xStartupLaunchConfigurator extends Karaf2xStartupLaunchConfigurator {
 	
 	public Fuse6xStartupLaunchConfigurator(IServer server) throws CoreException {
 		super(server);
 	}
 	
-	/* (non-Javadoc)
-	 * @see org.fusesource.ide.server.karaf.core.server.subsystems.BaseKarafStartupLaunchConfigurator#isSupportedRuntimeVersion(java.lang.String)
-	 */
 	@Override
 	protected boolean isSupportedRuntimeVersion(String version) {
 		return !Strings.isBlank(version) && version.startsWith(FuseToolingConstants.FUSE_VERSION_6X);
 	}
 	
 	@Override
-	protected String getVMArguments(String karafInstallDir, String endorsedDirs, String extDirs) {
-		// removed derby, introduced 2 new XX:(Max)PermSize parameters
-		StringBuilder vmArguments = new StringBuilder();
-		
-		vmArguments.append("-Xms128M");
-		vmArguments.append(SPACE + "-Xmx512M");
-		vmArguments.append(SPACE + "-XX:+UnlockDiagnosticVMOptions");
-		vmArguments.append(SPACE + "-XX:+UnsyncloadClass");
-		vmArguments.append(SPACE + "-XX:PermSize=16M -XX:MaxPermSize=128M ");
-		vmArguments.append(SPACE + "-server ");
-		vmArguments.append(SPACE + "-Dcom.sun.management.jmxremote");
-		vmArguments.append(SPACE + "-Djava.endorsed.dirs=" + QUOTE + endorsedDirs + QUOTE);
-		vmArguments.append(SPACE + "-Djava.ext.dirs=" + QUOTE + extDirs + QUOTE);
-		vmArguments.append(SPACE + "-Dkaraf.startLocalConsole=false");
-		vmArguments.append(SPACE + "-Dkaraf.startRemoteShell=true");
-		vmArguments.append(SPACE + "-Dkaraf.home=" + QUOTE + karafInstallDir + QUOTE); 
-		vmArguments.append(SPACE + "-Dkaraf.base=" + QUOTE + karafInstallDir + QUOTE);
-		vmArguments.append(SPACE + "-Dkaraf.instances=" + QUOTE + karafInstallDir + SEPARATOR + "instances" + QUOTE);
-		vmArguments.append(SPACE + "-Dkaraf.data=" + QUOTE + karafInstallDir + SEPARATOR + "data" + QUOTE);
-		vmArguments.append(SPACE + "-Dkaraf.etc=" + QUOTE + karafInstallDir + SEPARATOR + "etc" + QUOTE);
-		vmArguments.append(SPACE + "-Djava.io.tmpdir=" + QUOTE + karafInstallDir + SEPARATOR + "data" + SEPARATOR + "tmp" + QUOTE);
-		vmArguments.append(SPACE + "-Djava.util.logging.config.file=" + QUOTE + karafInstallDir + SEPARATOR + "etc" + SEPARATOR + "java.util.logging.properties" + QUOTE);
-		
-		return vmArguments.toString();
+	public String getVMArguments(String karafInstallDir, String endorsedDirs, String extDirs) {
+		StringBuilder sb = new StringBuilder(super.getVMArguments(karafInstallDir, endorsedDirs, extDirs));
+		sb.append(SPACE + "-XX:PermSize=16M -XX:MaxPermSize=128M");
+		return sb.toString();
```

Comment:
I don't think that this is easier to read than what I had here before.

Reply:
it allows to see which parameters are added for/since a specific version